frontend/appflowy_tauri/src/appflowy_app/components/_shared/image_upload/UploadImage.tsx

```diff
@@ -1,7 +1,109 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import Button from '@mui/material/Button';
+import { useTranslation } from 'react-i18next';
+import CloudUploadIcon from '@mui/icons-material/CloudUploadOutlined';
+import { notify } from '$app/components/_shared/notify';
+import { isTauri } from '$app/utils/env';
 
-export function UploadImage() {
-  return <div></div>;
+const MAX_IMAGE_SIZE = 10 * 1024 * 1024; // 10MB
+
+export function UploadImage({ onDone }: { onDone?: (url: string) => void }) {
+  const { t } = useTranslation();
+
+  const checkTauriFile = useCallback(
+    async (url: string) => {
+      const filename = url.split('/').pop();
```

Comment:
is there no built-in function to get the file name? relying on split could lead to issues if the file name includes a '/'.